### PR TITLE
Optimize cuckoo filter, and remove redundant config parameters.

### PR DESCRIPTION
**Pull request by iwasaki-kenta**
cuckoo, ledger, protocol, mod: optimize cuckoo filter to reduce number of rounds of hashing required, utilize jenkins hashing, and to improve marshal/unmarshaling speeds
conf, cmd/wavelet: remove unnecessary config parameters for bloom filter params

### CODE REVIEW

File: internal/cuckoo/filter.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math/rand"
+)
+
+const (
+	BucketSize           = 4
+	NumBuckets           = 524288
```
Comment (a-urth) [inline internal/cuckoo/filter.go]: why number of buckets is fixed? shouldn't it be dependent on number of store values?

File: internal/cuckoo/filter_test.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
+package cuckoo
```
Comment (a-urth) [inline internal/cuckoo/filter_test.go]: we use build tags to distinguish unit and integration tests

File: internal/cuckoo/filter.go (hunk 1 above)
Comment (iwasaki-kenta) [reply internal/cuckoo/filter.go]: It's a fixed parameter actually - the number of buckets should be consistent across all nodes.

File: internal/cuckoo/filter_test.go (hunk 2 above)
Comment (iwasaki-kenta) [reply internal/cuckoo/filter_test.go]: Yep, will fix.

**Pull request merged**